cli/mender-artifact/modify.go

```diff
@@ -46,7 +46,12 @@ func modifyArtifact(c *cli.Context) (err error) {
 		return cli.NewExitError("File ["+c.Args().First()+"] does not exist.", 1)
 	}
 
-	image, err := virtualImage.Open(comp, privateKey, c.Args().First())
+	var image VPImage
+	if c.String("compression") != "" {
+		image, err = virtualImage.Open(privateKey, c.Args().First(), comp)
+	} else {
+		image, err = virtualImage.Open(privateKey, c.Args().First())
+	}
```

Comment:
Not pretty, but it was the least invasive way I found around this issue

Reply:
Nope, because that is the compressor is there acquired from the `global` scope variable for `compression`, which defaults to `gzip`.

Reply:
:+1: